lib/wire-protocol.d.ts

```diff
@@ -87,12 +122,7 @@ export declare class ServiceError extends Error {
    * where message does not suffice.
    */
   readonly summary?: string;
-  constructor(error: {
-    code: string;
-    message: string;
-    httpStatus: number;
-    summary?: string;
-  });
+  constructor(failure: QueryFailure, httpStatus: number);
```

Comment:
```suggestion
  constructor(failure: QueryError, httpStatus: number);
```

Reply:
`QueryFailure` comes from the openAPI spec too. So I've followed a convention of using "failure" for the whole payload. The error being the contents of the `error` field in that payload.

Reply:
i see. yeah let's go with that then.